### PR TITLE
Fix the issue (#2102): Add cast to float32 from float16 into Stable D…

### PR DESCRIPTION
**Pull request by y-vectorfield-fj**
# What does this PR do?

This PR fixed  Issue #2102 (InvalidArgumentError occurred when I implemeted image generation using mixed float16) .

Fixes # (issue)
#2102

# Who can review?
@ianstenbit 
@jbischof 
Someone who know this problem.

### CODE REVIEW

File: keras_cv/models/stable_diffusion/stable_diffusion.py [hunk 1]
```diff
@@ -235,6 +235,8 @@ def generate_image(
                 + unconditional_guidance_scale * (latent - unconditional_latent)
             )
             a_t, a_prev = alphas[index], alphas_prev[index]
+            if latent.dtype == "float16":
```
Comment (ianstenbit) [inline keras_cv/models/stable_diffusion/stable_diffusion.py:238]: I think we should be able to unconditionally cast latent to the dtype of latent_prev, e.g. `ops.cast(latent, latent_prev.dtype)`.

Is there a reason that won't work?

**Review by ianstenbit — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/models/stable_diffusion/stable_diffusion.py (hunk 1 above)
Comment (y-vectorfield) [reply keras_cv/models/stable_diffusion/stable_diffusion.py:238]: Thank you very much for your comment.
I think you are right!
I will try your advice and modify the code.

Comment (y-vectorfield-fj) [reply keras_cv/models/stable_diffusion/stable_diffusion.py:238]: Thank you very much for your comment.
I think you are right!
I will try your advice and modify the code.

Comment (y-vectorfield-fj) [reply keras_cv/models/stable_diffusion/stable_diffusion.py]: @ianstenbit 
I tried your suggestion and modified the code!
Thank you!!
If there is no problem, please approve my modification.

**Pull request merged**